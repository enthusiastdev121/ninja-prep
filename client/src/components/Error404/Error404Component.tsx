/*
404 Not Found pages when user goes to route that does not exist
*/

import React from 'react'
import  Error404Desktop  from './Error404Desktop'
import  Error404Tablet  from './Error404Tablet'
import  Error404Mobile  from './Error404Mobile'
import { Desktop, Tablet, Mobile } from '../MediaQuery/DeviceSizes'

const Error404Component = () => (
    <div>
        <Desktop >
            <Error404Desktop />
        </Desktop>
        <Tablet>
            <Error404Tablet />
        </Tablet>
        <Mobile>
            <Error404Mobile />
        </Mobile>
    </div>
)

export default Error404Component