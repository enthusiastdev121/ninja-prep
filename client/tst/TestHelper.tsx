import React from 'react'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
const history = createBrowserHistory()

interface DEVICE_WIDTHS {
    readonly DESKTOP: number
    readonly MOBILE: number
    readonly TABLET: number
}

//Widths in Pixels. Aligns with Media Query widths
export const DEVICE_TYPES: DEVICE_WIDTHS = {
    DESKTOP: 1000,
    MOBILE: 750,
    TABLET: 800
}

export const routeWrapper = (children: React.ReactChild) => {
    return <Router history={history}>{children}</Router>
}
