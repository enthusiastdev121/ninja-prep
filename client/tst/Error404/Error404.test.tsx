import React from 'react';

import Error404Component from 'components/Error404';
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {Context as ResponsiveContext} from 'react-responsive';

import {DEVICE_TYPES, routeWrapper} from '../TestHelper';

test.each(Object.entries(DEVICE_TYPES))(
  'renders successfully on %p',
  (deviceType: string, deviceWidth: number) => {
    const wrappingComponent = ResponsiveContext.Provider;
    const wrappingComponentProps = {value: {width: deviceWidth}};
    const mountProps = {wrappingComponent, wrappingComponentProps};

    const wrapper = shallow(routeWrapper(<Error404Component />), mountProps);
    expect(wrapper).toBeDefined();
  },
);

test.each(Object.entries(DEVICE_TYPES))(
  'matches snapshot on %p',
  (deviceType: string, deviceWidth: number) => {
    const wrappingComponent = ResponsiveContext.Provider;
    const wrappingComponentProps = {value: {width: deviceWidth}};
    const mountProps = {wrappingComponent, wrappingComponentProps};

    const wrapper = mount(routeWrapper(<Error404Component />), mountProps);
    expect(toJson(wrapper)).toMatchSnapshot();
  },
);
