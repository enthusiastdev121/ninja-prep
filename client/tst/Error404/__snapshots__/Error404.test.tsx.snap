// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot on "DESKTOP" 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Error404>
    <styled.div>
      <div
        className="sc-bdnxRM eeGZGd"
      >
        <styled.h1>
          <h1
            className="sc-hKFxyN dIIkaJ display-1 mt-5"
          >
             
            404
             
          </h1>
        </styled.h1>
        <styled.div>
          <div
            className="sc-eCApnc fLcWfp mt-3"
          >
             
            Oops!
             
          </div>
        </styled.div>
        <styled.div>
          <div
            className="sc-eCApnc fLcWfp mt-3"
          >
             
            We can't seem to find the page you are looking for...
             
          </div>
        </styled.div>
        <styled.div>
          <div
            className="sc-eCApnc fLcWfp mt-3"
          >
            <styled.button>
              <button
                className="sc-dlnjwi mt-5 btn-primary btn"
              >
                <Styled(Link)
                  to="/"
                >
                  <Link
                    className="sc-gtsrHT iZmkcW"
                    to="/"
                  >
                    <LinkAnchor
                      className="sc-gtsrHT iZmkcW"
                      href="/"
                      navigate={[Function]}
                    >
                      <a
                        className="sc-gtsrHT iZmkcW"
                        href="/"
                        onClick={[Function]}
                      >
                        Back to home
                      </a>
                    </LinkAnchor>
                  </Link>
                </Styled(Link)>
              </button>
            </styled.button>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </Error404>
</Router>
`;

exports[`matches snapshot on "MOBILE" 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Error404>
    <styled.div>
      <div
        className="sc-bdnxRM eeGZGd"
      >
        <styled.h1>
          <h1
            className="sc-hKFxyN dIIkaJ display-1 mt-5"
          >
             
            404
             
          </h1>
        </styled.h1>
        <styled.div>
          <div
            className="sc-eCApnc fLcWfp mt-3"
          >
             
            Oops!
             
          </div>
        </styled.div>
        <styled.div>
          <div
            className="sc-eCApnc fLcWfp mt-3"
          >
             
            We can't seem to find the page you are looking for...
             
          </div>
        </styled.div>
        <styled.div>
          <div
            className="sc-eCApnc fLcWfp mt-3"
          >
            <styled.button>
              <button
                className="sc-dlnjwi mt-5 btn-primary btn"
              >
                <Styled(Link)
                  to="/"
                >
                  <Link
                    className="sc-gtsrHT iZmkcW"
                    to="/"
                  >
                    <LinkAnchor
                      className="sc-gtsrHT iZmkcW"
                      href="/"
                      navigate={[Function]}
                    >
                      <a
                        className="sc-gtsrHT iZmkcW"
                        href="/"
                        onClick={[Function]}
                      >
                        Back to home
                      </a>
                    </LinkAnchor>
                  </Link>
                </Styled(Link)>
              </button>
            </styled.button>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </Error404>
</Router>
`;

exports[`matches snapshot on "TABLET" 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Error404>
    <styled.div>
      <div
        className="sc-bdnxRM eeGZGd"
      >
        <styled.h1>
          <h1
            className="sc-hKFxyN dIIkaJ display-1 mt-5"
          >
             
            404
             
          </h1>
        </styled.h1>
        <styled.div>
          <div
            className="sc-eCApnc fLcWfp mt-3"
          >
             
            Oops!
             
          </div>
        </styled.div>
        <styled.div>
          <div
            className="sc-eCApnc fLcWfp mt-3"
          >
             
            We can't seem to find the page you are looking for...
             
          </div>
        </styled.div>
        <styled.div>
          <div
            className="sc-eCApnc fLcWfp mt-3"
          >
            <styled.button>
              <button
                className="sc-dlnjwi mt-5 btn-primary btn"
              >
                <Styled(Link)
                  to="/"
                >
                  <Link
                    className="sc-gtsrHT iZmkcW"
                    to="/"
                  >
                    <LinkAnchor
                      className="sc-gtsrHT iZmkcW"
                      href="/"
                      navigate={[Function]}
                    >
                      <a
                        className="sc-gtsrHT iZmkcW"
                        href="/"
                        onClick={[Function]}
                      >
                        Back to home
                      </a>
                    </LinkAnchor>
                  </Link>
                </Styled(Link)>
              </button>
            </styled.button>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </Error404>
</Router>
`;
